### PR TITLE
grpc: create docs page for tutorial

### PR DESCRIPTION
**Pull request by ggilmore**
This PR adds a docs page that advertises the [gRPC tutorial](https://github.com/sourcegraph/sourcegraph/blob/main/internal/grpc/example/doc.go) that we added in https://github.com/sourcegraph/sourcegraph/pull/58790.

### CODE REVIEW

**Comment by ggilmore** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #51** <a href="https://app.graphite.dev/github/pr/sourcegraph/docs/51?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/docs/51?utm_source=stack-comment).

File: docs/dev/background-information/grpc_tutorial.mdx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# gRPC
+
+As of Sourcegraph `5.3.X`, [gRPC](https://grpc.io/about/) has supplanted REST as our default mode of communication between our microservices for our internal APIs.
+
+> NOTE: An "internal" API is one that's solely used for intra-service communication/RPCs (think `searcher` fetching an archive from `gitserver`). Internal APIs don't include things like the graphQL API that external people can use (including our web interface).
```
Comment (jtibshirani) [inline docs/dev/background-information/grpc_tutorial.mdx:5]: Super tiny comment, usually GraphQL is capitalized

File: docs/dev/index.mdx [hunk 2]
```diff
@@ -67,7 +67,7 @@ Clarification and discussion about key concepts, architecture, and development s
     - [Testing web code](/dev/background-information/testing_web_code)
 - [Building p4-fusion](/dev/background-information/build_p4_fusion)
 - [The `gitserver` API](/dev/background-information/gitserver-api)
-- [Using gRPC alongside REST for internal APIs](/dev/background-information/gRPC_internal_api)
+- [gRPC Tutorial](/dev/background-information/grpc_tutorial)
```
Comment (jtibshirani) [inline docs/dev/index.mdx:70]: Not actually related to this PR, but it'd be nice if there were more structure here, for example having a section for all things related to "backend platform" development.

**Review by jtibshirani — APPROVED** [review]
Looks good! Looking forward to working through the tutorial soon.

**Review by eseliger — APPROVED** [review]

File: docs/dev/background-information/grpc_tutorial.mdx (hunk 1 above)
Comment (MaedahBatool) [inline docs/dev/background-information/grpc_tutorial.mdx:5]:
```suggestion
<Callout type="note">An "internal" API is one that's solely used for intra-service communication/RPCs (think `searcher` fetching an archive from `gitserver`). Internal APIs don't include things like the graphQL API that external people can use (including our web interface).</Callout>
```

We have a component for Note callout. Please use this.

File: docs/dev/background-information/grpc_tutorial.mdx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+## gRPC Tutorial
+
+The [`internal/grpc/example`](https://github.com/sourcegraph/sourcegraph/tree/main/internal/grpc/example) package in the [sourcegraph/sourcegraph monorepo](https://github.com/sourcegraph/sourcegraph) contains a simple, runnable example of a gRPC service and client. It is a good starting point for understanding how to write a gRPC service that covers following topics:
```
Comment (MaedahBatool) [inline docs/dev/background-information/grpc_tutorial.mdx:9]:
```suggestion
The [`internal/grpc/example`](https://github.com/sourcegraph/sourcegraph/tree/main/internal/grpc/example) package in the [sourcegraph/sourcegraph monorepo](https://github.com/sourcegraph/sourcegraph) contains a simple, runnable example of a gRPC service and client. It is a good starting point for understanding how to write a gRPC service that covers the following topics:
```

**Review by MaedahBatool — CHANGES_REQUESTED** [review]
Small fixes.

**Review by MaedahBatool — APPROVED** [review]

File: docs/dev/index.mdx (hunk 2 above)
Comment (ggilmore) [reply docs/dev/index.mdx:70]: We can revisit this in a separate change

**Pull request merged**